src/components/WeatherWidgetDropdown.vue

```diff
@@ -0,0 +1,207 @@
+
+const searchValue = ref('');
+
+defineProps<{ selectedLocationKey: PlaceInfo | null }>();
+
+const emit = defineEmits<{
+  (e: 'handleLocationKeyChange', autocompleteData: PlaceInfo | null): void;
+}>();
+
+const {
+  data: autocompleteData,
+  error: autocompleteError,
+  isLoading: isLoadingAutocomplete,
+  fetchData: fetchAutocompleteResults,
+} = useFetch<AutocompleteMapperOutput>(
+  `${API_URL}${ENDPOINTS.autocomplete}`,
+  mapAutocompleteData
+);
+
+const loadingMessage = computed(() =>
+  isLoadingAutocomplete.value ? 'Loading...' : ''
+);
+
+const noResultsMessage = computed(() => {
+  if (
+    !loadingMessage.value &&
+    (!autocompleteData.value || autocompleteData.value.length === 0)
+  ) {
+    return 'No results.';
+  }
+});
+
+const statusMessages = computed(() => {
+  return [
+    loadingMessage.value,
+    autocompleteError.value,
+    noResultsMessage.value,
+  ].filter(Boolean) as (string | Error)[];
+});
+
+const debouncedFetchAutocompleteResults = debounce(fetchAutocompleteResults);
+
+const handleInputChange = () => {
+  if (!searchValue.value) return;
+
+  debouncedFetchAutocompleteResults('GET', {
+    params: { q: searchValue.value },
+  });
+};
+
+const handlePlaceChange = (place: PlaceInfo | null) => {
+  searchValue.value = place ? `${place.localizedName}, ${place.country}` : '';
+  emit('handleLocationKeyChange', place);
+};
+</script>
+
+<style scoped lang="scss">
+.weather-widget-dropdown {
+  $dropdown-max-height: 240px;
+  $input-width: 160px;
+
+  display: flex;
+  gap: $space-xs;
+  margin-left: auto;
+  width: min-content;
+
+  &__input-wrapper {
+    position: relative;
+  }
+
+  &__input,
+  &__reset-btn {
+    border: $border-primary;
+    outline: none;
+    border-radius: $radius-sm;
+    padding: $p-xs;
+    background-color: $clr-primary-dark;
+    cursor: pointer;
+    color: $clr-text;
+    font-size: $fs-sm;
+    transition: all $transition-duration ease-out;
+
+    &:hover {
+      background-color: $clr-primary-light;
+    }
+  }
+
+  &__input {
+    width: $input-width;
+
+    &::placeholder {
+      color: $clr-text;
+      text-align: center;
+      transition: all $transition-duration ease-out;
+    }
+
+    &:focus::placeholder {
+      color: $clr-text-muted;
+    }
+
+    &:focus,
+    &:not(:placeholder-shown) {
+      border-radius: $radius-sm $radius-sm 0 0;
+      width: $input-width * 2;
+      background-color: $clr-primary-light;
+      cursor: text;
+    }
+  }
+
+  &__reset-btn {
+    overflow: hidden;
+  }
+
+  &__input-wrapper:active &__input,
+  &__input:focus {
+    & ~ ul {
+      max-height: $dropdown-max-height;
+      border-color: $clr-primary-very-light;
+
+      & li {
+        opacity: 1;
+      }
+    }
+  }
+
+  &__list {
+    position: absolute;
+    top: 100%;
+    left: 0%;
+    width: 100%;
+    max-height: 0;
+    overflow-y: scroll;
+    list-style: none;
+    border: $border-primary;
+    border-top: none;
+    border-color: transparent;
```

Comment:
Since you're overwriting `$border-primary` I don't see much sense in using `$border-primary` here. Let's just be explicit and write everything in `border` property.